c/datasetjson.c

```diff
@@ -1630,6 +1640,395 @@ void respondWithDataset(HttpResponse* response, char* absolutePath, int jsonMode
+    int i;
+    for(i=0; i<52; i++)
+    { 
+        zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,"%c",*(dsn_member.dsn+i));
+    }
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,">\n");
+
+    memcpy(rname_parm.value, &dsn_member, sizeof(dsn_member) ); /* copy in dsn+membername */
+    rname_parm.length = 52;
+
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,
+      "ZSS1809I %s rname_parm\n", __FUNCTION__);
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,"<"); 
+    for(i=0; i<52+sizeof(unsigned char); i++)
+    { 
+        zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,"%x",*(rname_parm.value+i-sizeof(unsigned char)));
+    }
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,">\n");
+
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,"<"); 
+    for(i=0; i<52; i++)
+    { 
+        zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,"%c",*(rname_parm.value+i));
+    }
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,">\n");
+    
+/* test lock */
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG, 
+      "ZSS1826I %s Try exclusive Lock\n", __FUNCTION__);
+    
+/*    printf("TryExclusiveLock\n");  waits forever? CONTENTIONACT=FAIL */
+    lockRC = isgenqTryExclusiveLock(
+        &MajorQNAME, 
+        &rname_parm, 
+        ISGENQ_SCOPE_SYSTEMS, 
+        &lockToken, 
+        &lockRSN);   
+
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG, 
+      "ZSS1757I %s lockRC = %d, lockRSN = %x\n", __FUNCTION__ , lockRC , lockRSN );  
+
+
+    if ( lockRC != 0){
+
+
+
+        if (lockRC == 4)
+        {
+            zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_WARNING, 
+              "ZSS1773E %s isgenqTryExclusiveLock Unable to obtain exclusive access to Dataset or member\n"
+              , __FUNCTION__);
+            respondWithError(response,HTTP_STATUS_RESOURCE_CONFLICT,
+              "Unable to obtain exclusive access to Dataset or member");
+            return;
+        }
+        else        {
+            zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG, 
+              "ZSS1779E %s isgenqTryExclusiveLock failed, RC = %d, reason code = %08X, %04X\n" , __FUNCTION__, 
+              lockRC , lockRSN, lockRSN & 0xFFFF);
+            respondWithError(response,HTTP_STATUS_INTERNAL_SERVER_ERROR,
+              "isgenqTryExclusiveLock failed");
+            return;
+        }
+    }
+    else{
+        zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG, 
+              "ZSS1789I %s Lock is available\n", __FUNCTION__);
+    }
+        
+/* test lock end */
+
+    zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,
+      "ZSS1793I %s GetExclusiveLock\n", __FUNCTION__);
+
+    lockRC = isgenqGetExclusiveLock( 
+        &MajorQNAME, 
+        &rname_parm, 
+        ISGENQ_SCOPE_SYSTEMS, 
+        &lockToken, 
+        &lockRSN);   
+    /* This function WAITs in case of contention.  This should not be a problem because it's preceeded by isgenqTryExclusiveLock */
```

Comment:
No, this is a problem because someone can acquire that lock in between the calls.